extensions/mssql/src/features.ts

```diff
@@ -86,7 +91,40 @@ export class AccountFeature implements StaticFeature {
 
 		let params: contracts.RequestSecurityTokenResponse = {
 			accountKey: JSON.stringify(account.key),
-			token: securityToken.token
+			token: securityToken.token,
+			expiresOn: securityToken.expiresOn
+		};
+
+		return params;
+	}
+
+	protected async refreshToken(request: contracts.RefreshTokenNotificationParams) {
+
+		// find account
+		const accountList = await azdata.accounts.getAllAccounts();
+		let account: azdata.Account;
+		accountList.forEach(element => {
+			if (element.key.accountId === request.accountId) {
+				account = element;
+			}
+		});
+
+		// find tenant
+		const tenant = account.properties.tenants.find((tenant: azurecore.Tenant) => request.authority.includes(tenant.id));
+		const unauthorizedMessage = localize('mssql.insufficientlyPrivelagedAzureAccount', "The configured Azure account for {0} does not have sufficient permissions for Azure Key Vault to access a column master key for Always Encrypted.", account.key.accountId);
```

Comment:
Put this in a constants file

But also - why is this being specific to Always Encrypted? Tokens are used for more than that...